lib/trento/infrastructure/commanded/event_handlers/database_deregistration_event_handler.ex

```diff
@@ -0,0 +1,49 @@
+defmodule Trento.Infrastructure.Commanded.EventHandlers.DatabaseDeregistrationEventHandler do
+  @moduledoc """
+  This event handler is responsible to forward deregistration commands to the SAP systems
+  related to a deregistered database
+  """
+
+  use Commanded.Event.Handler,
+    application: Trento.Commanded,
+    name: "database_deregistration_event_handler"
+
+  alias Trento.Databases.Events.DatabaseDeregistered
+  alias Trento.Repo
+  alias Trento.SapSystems.Commands.DeregisterSapSystem
+  alias Trento.SapSystems.Projections.DatabaseReadModel
+
+  import Ecto.Query, only: [from: 2]
+
+  require Logger
+
+  def handle(
+        %DatabaseDeregistered{database_id: database_id, deregistered_at: deregistered_at},
+        _metadata
+      ) do
+    database =
```

Comment:
Could we deconstruct here? `%{sap_systems: sap_systems} = `

Reply:
done